### DISCUSSION TITLE
Unwanted downloads after a successful one [octane]

### DISCUSSION THREAD

**Discussion by agustinprod** [livewire/livewire#3492] [7. Bugs (archived)]
Hi,
we are having an issue with livewire downloads, the problem only happens when livewire is used in conjunction with Octane. For some reason, when the first download is made through Livewire, all the subsequent request came with the `effect['download']` already set, like something on the response object is cached between requests. The only hotfix we found so far is to regenerate the component ID after a download.

The SupportFileDownloads has this piece of code:

`Livewire::listen('action.returned', function ($component, $action, $returned) {`

The returned object is always the same after a successful download, so maybe the response object should be cleared on octane next request hooks.

Also, the $downloadsById probably should require to be cleared if the current response isnt a file download, but not totally sure about that.

**Comment by joshhanley** [comment]
@killgt there are some issues with Livewire and Octane, have a look at #3043. I believe this would probably be a similar issue.

**Comment by davidhsianturi** [comment]
I'm also having the same problem.
when the first download completes and the component return/restores the download again after a refresh or revisit of the page.

@killgt how do you regenerate the component ID after a download?

**Reply by agustinprod** [reply]
Add this before sending the download
        $this->id = Str::random();

**Reply by davidhsianturi** [reply]
thanks

**Comment by joshhanley** [comment]
This has been fixed in Livewire #3987 and laravel/octane#400 PR's and they have been merged, so this should be fixed in the next releases of Octane and Livewire.

Hope this helps!
